bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/eGateHandler.java

```diff
@@ -0,0 +1,344 @@
+        // always return true even after the socket is closed
+        // http://stackoverflow.com/questions/10163358/
+        return egateSocket.isConnected() && !egateSocket.isClosed();
+    }
+
+    /**
+     * Possible Instructions are:
+     * FssTransmit 1 Kommandoabsetzung (Controller > eGate > Dominoswiss)
+     * FssReceive 2 Empfangenes Funkpaket (Dominoswiss > eGate > Controller)
+     *
+     */
+
+    public void tiltUp(String id) {
+        for (int i = 0; i < 3; i++) {
+            pulseUp(id);
+            try {
+                Thread.sleep(150); // sleep to not confuse the blinds
+            } catch (InterruptedException e) {
+                logger.error("EGate tiltUp error: {}", e.toString());
+            }
+        }
+    }
+
+    public void tiltDown(String id) {
+        for (int i = 0; i < 3; i++) {
+            pulseDown(id);
+            try {
+                Thread.sleep(150);// sleep to not confuse the blinds
+            } catch (InterruptedException e) {
+                logger.error("EGate tiltDown error: {} ", e.toString());
+            }
+        }
+    }
+
+    public void pulseUp(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=1;Priority=1;CheckNr=3415347;" + CR);
+    }
+
+    public void pulseDown(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=2;Priority=1;CheckNr=2764516;" + CR);
+    }
+
+    public void continuousUp(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=3;Priority=1;CheckNr=2867016;" + CR, 20000);
+    }
+
+    public void continuousDown(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=4;Priority=1;CheckNr=973898;" + CR, 20000);
+    }
+
+    public void stop(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=5;Priority=1;CheckNr=5408219;" + CR);
+    }
+
+    @SuppressWarnings("null")
+    public void registerBlind(String id, ThingUID uid) {
+        logger.debug("Registring Blind id {} with thingUID {}", id, uid);
+        registeredBlinds.put(id, uid);
+    }
+
+    /**
+     * Send a command to the eGate Server.
+     */
+
+    private void sendCommand(String command) {
+        sendCommand(command, SOCKET_TIMEOUT);
+    }
+
+    @SuppressWarnings("null")
+    private synchronized void sendCommand(String command, int timeout) {
+        logger.debug("EGate got command: {}", command);
+
+        if (!isConnected()) {
+            logger.debug("no connection to Dominoswiss eGate server when trying to send command, returning...");
+            return;
+        }
+
+        // Send plain string to eGate Server,
+        logger.debug("Sending command: {}", command);
+        try {
+
+            // BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(egateSocket.getOutputStream()));
+
+            egateSocket.setSoTimeout(SOCKET_TIMEOUT);
+            writer.write(command);
+            writer.flush();
+            logger.debug("Sent message");
```

Comment:
Can this message be replaced by using the debugger or by increasing the framework's log level? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging